build.gradle

```diff
@@ -355,44 +355,38 @@ project(':iceberg-pig') {
 project(':iceberg-runtime') {
   apply plugin: 'com.github.johnrengelman.shadow'
 
-  configurations {
-    shadow
-    compileOnly.extendsFrom shadow
-  }
-
-  shadowJar {
-    configurations = [project.configurations.shadow]
-    from(project.sourceSets.main.output)
-  }
-
-  tasks.build.dependsOn tasks.shadowJar
+  tasks.assemble.dependsOn tasks.shadowJar
   tasks.install.dependsOn tasks.shadowJar
   tasks.javadocJar.dependsOn tasks.shadowJar
 
   dependencies {
-    shadow project(':iceberg-api')
-    shadow project(':iceberg-common')
-    shadow project(':iceberg-core')
-    shadow project(':iceberg-parquet')
-    shadow project(':iceberg-spark')
-    shadow project(':iceberg-pig')
-    shadow project(':iceberg-hive')
-
-    shadow "org.apache.avro:avro:$avroVersion"
-    shadow "org.apache.parquet:parquet-avro:$parquetVersion"
+    compile project(':iceberg-api')
+    compile project(':iceberg-common')
+    compile project(':iceberg-core')
+    compile project(':iceberg-parquet')
+    compile project(':iceberg-spark')
+    compile project(':iceberg-pig')
+    compile project(':iceberg-hive')
+
+    compile "org.apache.avro:avro:$avroVersion"
+    compile "org.apache.parquet:parquet-avro:$parquetVersion"
+
   }
 
   publishing {
     publications {
       withType(IvyPublication) {
-        configurations {
-          'shadow' { extend 'runtime' }
-        }
+        from components.java
       }
     }
   }
 
   shadowJar {
+    dependencies {
+      exclude(dependency {
+        it.moduleGroup == 'org.apache.hadoop'
```

Comment:
AFAICT this just excludes that specific dependency not all of its transitive dependencies.
For example, we lets say hadoop-common [or hadoop-hdfs] brought in guava. Only hadoop-common will be excluded, not guava. 
@SurenNihalani  can u please verify if that is the case?
If it is then we would be better off by excluding the hadoop dependencies from directly the dependencies block of iceberg-runtime by excluding it from orc

Reply:
The reason why we used the `shadow` configuration was to pull in just the dependencies that we wanted to provide in the shaded Jar, not everything. That way, for libraries we know are going to be present at runtime, like Hadoop, Spark, or even SLF4J, we don't have to write specific exclusions. These exclusions get out of date really quickly.

I still want to have a set of dependencies that we include explicitly.

Reply:
Not sure I follow. Using shadow vs compile will not make a difference on the dependencies and transitive dependencies that gets bundled in the fat jar.

Are you saying that we should add each dependency explicitly and not their transitives? This could be possible but it would get very complex very quickly as we'd have to write all dependencies explicitly and exclude their transitives

Reply:
@rdblue, I don't think the name of configuration matters here. The way people express that distinction is by using the runtime configuration (Command F for Figure 2 on this page: https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_plugin_and_dependency_management). I decided to move away from shadow because the shadow plugin adds a configuration named "shadow" (https://imperceptiblethoughts.com/shadow/getting-started/#default-java-groovy-tasks) and users use that configuration to specify dependencies that they want to include (https://imperceptiblethoughts.com/shadow/configuration/#configuring-the-runtime-classpath). Overriding the name shadow made things confusing if you know specifics of shadow plugin. 

@rdsr , I'll confirm and update the PR